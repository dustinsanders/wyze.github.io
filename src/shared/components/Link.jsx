// @flow

import type { Style, VNode } from '../../types'
import { css, hover } from 'glamor'
import preact from 'preact'

type Props = {
  children?: VNode,
  styles?: Style,
  to: string,
}

const link: Style = css(
  {
    color: '#db0a5b',
    textDecoration: 'none',
    transitionDuration: '.5s',
  },
  hover({
    color: '#8a2be2',
  }),
)

const Link = ({ children, styles, to }: Props): VNode => (
  <a
    {...css(link, styles)}
    href={to}
    rel="noopener noreferrer"
    target="_blank"
  >
    {children}
  </a>
)

Link.defaultProps = {
  children: null,
  styles: {},
}

export default Link
