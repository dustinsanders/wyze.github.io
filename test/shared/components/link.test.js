// @flow

import Link from '../../../src/shared/components/Link'
import snapshot from '../../helpers/snapshot'

const link = {
  children: 'GitHub',
  to: '//github.com/wyze',
}

const linkWithStyles = {
  ...link,
  styles: {
    color: 'rebeccapurple',
  },
}

describe('<Link />', () => {
  it('renders', snapshot.bind(null, Link, link))
  it('renders style', snapshot.bind(null, Link, linkWithStyles))
})
