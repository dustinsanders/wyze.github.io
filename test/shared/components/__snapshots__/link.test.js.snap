// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders 1`] = `"<a data-css-1grb0d9 href=\\"//github.com/wyze\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">GitHub</a>"`;

exports[`<Link /> renders style 1`] = `"<a data-css-v7fx40 href=\\"//github.com/wyze\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">GitHub</a>"`;
